lib/Engine.php

```diff
@@ -106,7 +106,7 @@ public function generatePassword() {
 	}
 
 	/**
-	 * @param string $password
+	 * @param string|null $password
 	 * @param string $uid
```

Comment:
```suggestion
	 * @param string|null $uid
```
this can also be null?